crates/pop-cli/src/commands/call/parachain.rs

```diff
@@ -6,207 +6,222 @@ use clap::Args;
 use pop_parachains::{
 	construct_extrinsic, encode_call_data, find_extrinsic_by_name, find_pallet_by_name,
 	parse_chain_metadata, set_up_api, sign_and_submit_extrinsic, supported_actions, Action,
-	DynamicPayload, OnlineClient, Pallet, Param, SubstrateConfig,
+	DynamicPayload, Extrinsic, OnlineClient, Pallet, Param, SubstrateConfig,
 };
+use url::Url;
 
 const DEFAULT_URL: &str = "ws://localhost:9944/";
 const DEFAULT_URI: &str = "//Alice";
 
 #[derive(Args, Clone)]
 pub struct CallParachainCommand {
 	/// The pallet containing the extrinsic to execute.
-	#[clap(long, short)]
+	#[arg(long)]
 	pallet: Option<String>,
 	/// The extrinsic to execute within the chosen pallet.
-	#[clap(long, short)]
+	#[arg(long)]
 	extrinsic: Option<String>,
 	/// The extrinsic arguments, encoded as strings.
-	#[clap(long, num_args = 0..,)]
+	#[arg(long, num_args = 0..,)]
 	args: Vec<String>,
 	/// Websocket endpoint of a node.
-	#[clap(name = "url", short = 'u', long, value_parser, default_value = DEFAULT_URL)]
-	url: url::Url,
+	#[arg(short = 'u', long, value_parser)]
+	url: Option<Url>,
 	/// Secret key URI for the account signing the extrinsic.
 	///
 	/// e.g.
 	/// - for a dev account "//Alice"
 	/// - with a password "//Alice///SECRET_PASSWORD"
-	#[clap(name = "suri", long, short, default_value = DEFAULT_URI)]
-	suri: String,
+	#[arg(long)]
+	suri: Option<String>,
 	/// Automatically signs and submits the extrinsic without asking for confirmation.
-	#[clap(short('y'), long)]
+	#[arg(short('y'), long)]
 	skip_confirm: bool,
 }
 
 impl CallParachainCommand {
 	/// Executes the command.
 	pub(crate) async fn execute(mut self) -> Result<()> {
-		// Check if message specified via command line argument.
-		let prompt_to_repeat_call = self.extrinsic.is_none();
```

Comment:
This change makes sense. However, if the user has provided all the necessary fields, I don't think we should prompt them to repeat the call; this behavior should be reserved for the interactive UI.

For instance, if I execute the following extrinsic:
`pop call parachain --pallet Nfts --extrinsic mint --args "" --url ws://127.0.0.1:9944/ --suri //Alice,`
at the end, the UI continues to wait for a response, which feels unnecessary in this context (Think of a script using it).

We could address this by introducing a flag like `skip_confirm`, or by extending the logic we had before but verifying that all required fields are set instead of only the `extrinsic`. For example:
```rs
let prompt_to_repeat_call = self.pallet.is_none() 
    && self.extrinsic.is_none() 
    && self.args.is_some() 
    && self.url.is_some() 
    && self.suri.is_some();
```

Reply:
I'll add it